internal/controller/addons_controller.go

```diff
@@ -0,0 +1,81 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/kyma-project/helm-broker/internal"
+	"github.com/kyma-project/helm-broker/pkg/apis/addons/v1alpha1"
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var _ reconcile.Reconciler = &ReconcileAddonsConfiguration{}
+
+// ReconcileAddonsConfiguration reconciles a AddonsConfiguration object
+type ReconcileAddonsConfiguration struct {
+	log logrus.FieldLogger
+	client.Client
+
+	common commonReconciler
+}
+
+// NewReconcileAddonsConfiguration returns a new reconcile.Reconciler
+func NewReconcileAddonsConfiguration(mgr manager.Manager, addonGetterFactory addonGetterFactory, chartStorage chartStorage, addonStorage addonStorage, brokerFacade brokerFacade, docsProvider docsProvider, brokerSyncer brokerSyncer, tmpDir string, log logrus.FieldLogger) reconcile.Reconciler {
+	return &ReconcileAddonsConfiguration{
+		log:    log.WithField("controller", "addons"),
+		Client: mgr.GetClient(),
+
+		common: newControllerCommon(mgr.GetClient(), addonGetterFactory, addonStorage, chartStorage, docsProvider, brokerSyncer, brokerFacade, path.Join(tmpDir, "addon-loader-dst"), log),
+	}
+}
+
+// Reconcile reads that state of the cluster for a AddonsConfiguration object and makes changes based on the state read
+func (r *ReconcileAddonsConfiguration) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	addon := &v1alpha1.AddonsConfiguration{}
+	err := r.Get(context.TODO(), request.NamespacedName, addon)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	r.common.SetWorkingNamespace(addon.Namespace)
+	commonAddon := &internal.CommonAddon{
+		Meta:   addon.ObjectMeta,
+		Spec:   addon.Spec.CommonAddonsConfigurationSpec,
+		Status: addon.Status.CommonAddonsConfigurationStatus,
+	}
+
+	return r.common.Reconcile(commonAddon, fmt.Sprintf("AddonsConfiguration `%s/%s`", addon.Name, addon.Namespace))
```

Comment:
`cluster_addons_controller` uses `commonAddon.Meta.Name` maybe will be more consistent use here `commonAddon.Meta.X` or there use `addon.X`?

Reply:
good point, I will use `commonAddon.Meta.X` in both controllers